app/upgrades/v17/upgrades.go

```diff
@@ -6,17 +6,45 @@ package v17
 import (
 	sdk "github.com/cosmos/cosmos-sdk/types"
 	"github.com/cosmos/cosmos-sdk/types/module"
+	authkeeper "github.com/cosmos/cosmos-sdk/x/auth/keeper"
+	bankkeeper "github.com/cosmos/cosmos-sdk/x/bank/keeper"
 	upgradetypes "github.com/cosmos/cosmos-sdk/x/upgrade/types"
+	"github.com/ethereum/go-ethereum/common"
+	erc20precompile "github.com/evmos/evmos/v16/precompiles/erc20"
+	"github.com/evmos/evmos/v16/utils"
+	erc20keeper "github.com/evmos/evmos/v16/x/erc20/keeper"
 )
 
 // CreateUpgradeHandler creates an SDK upgrade handler for v17.0.0
 func CreateUpgradeHandler(
 	mm *module.Manager,
 	configurator module.Configurator,
+	ak authkeeper.AccountKeeper,
+	bk bankkeeper.Keeper,
+	erck erc20keeper.Keeper,
 ) upgradetypes.UpgradeHandler {
 	return func(ctx sdk.Context, _ upgradetypes.Plan, vm module.VersionMap) (module.VersionMap, error) {
 		logger := ctx.Logger().With("upgrade", UpgradeName)
 
+		var wrappedContractAddr common.Address
+		switch {
+		case utils.IsMainnet(ctx.ChainID()):
+			wrappedContractAddr = common.HexToAddress(erc20precompile.WEVMOSContractMainnet)
+		case utils.IsTestnet(ctx.ChainID()):
+			wrappedContractAddr = common.HexToAddress(erc20precompile.WEVMOSContractTestnet)
+		default:
+			logger.Error("unexpected chain id", "chain-id", ctx.ChainID())
+		}
+
+		// Execute the conversion for all Cosmos native ERC20 token pairs to use the ERC20 EVM extension.
```

Comment:
Not very clear this comment, what is a Cosmos native erc20?

Reply:
It's an ERC-20 token pair for a native Cosmos Coin (like `uosmo`) - do you have a suggestion how to make it clearer? 🤔

Reply:
Yeah, I would just write what you just said ahah
```suggestion
		// Execute the conversion for all ERC20 token pairs for native Cosmos coins to use the ERC20 EVM extension.
```